include/deal.II/grid/grid_tools.h

```diff
@@ -1727,6 +1727,9 @@ namespace GridTools
                      + cell->face(face)->vertex(2) + cell->face(face)->vertex(3)) / 4.0;
               }
       }
+
+    // Make sure FEValues notices that the mesh has changed
+    triangulation.signals.post_refinement();
```

Comment:
I guess we should go with a new signal, I agree with you. At first sight, the only place that should connect to that signal is `fe_values.h`, the rest is in the triangulation or DoFHandler implementation which will not need to know that I guess...